table/neuron/table.cpp

```diff
@@ -0,0 +1,409 @@
+    static void _check_table_thread(Memb_list* _ml, size_t id, Datum* _ppvar, Datum* _thread, NrnThread* _nt, int _type, _nrn_model_sorted_token const& _sorted_token)
+{
+        _nrn_mechanism_cache_range _lmr{_sorted_token, *_nt, *_ml, _type};
+        auto inst = make_instance_tbl(_lmr);
+        check_sigmoid1_tbl(&_lmr, inst, id, _ppvar, _thread, _nt);
+    }
+
+
+    /** connect global (scalar) variables to hoc -- */
+    static DoubScal hoc_scalar_double[] = {
+        {"k_tbl", &tbl_global.k},
+        {"d_tbl", &tbl_global.d},
+        {"usetable_tbl", &tbl_global.usetable},
+        {nullptr, nullptr}
+    };
+
+
+    /** connect global (array) variables to hoc -- */
+    static DoubVec hoc_vector_double[] = {
+        {nullptr, nullptr, 0}
+    };
+
+
+    /* declaration of user functions */
+    static void _hoc_sigmoid1(void);
+    static double _npy_sigmoid1(Prop*);
+
+
+    /* connect user functions to hoc names */
+    static VoidFunc hoc_intfunc[] = {
+        {"setdata_tbl", _hoc_setdata},
+        {"sigmoid1_tbl", _hoc_sigmoid1},
+        {0, 0}
+    };
+    static NPyDirectMechFunc npy_direct_func_proc[] = {
+        {"sigmoid1", _npy_sigmoid1},
+    };
+    static void _hoc_sigmoid1(void) {
+        double _r{};
+        Datum* _ppvar;
+        Datum* _thread;
+        NrnThread* _nt;
+        Prop* _local_prop = _prop_id ? _extcall_prop : nullptr;
+        _nrn_mechanism_cache_instance _ml_real{_local_prop};
+        auto* const _ml = &_ml_real;
+        size_t const id{};
+        _ppvar = _local_prop ? _nrn_mechanism_access_dparam(_local_prop) : nullptr;
+        _thread = _extcall_thread.data();
+        _nt = nrn_threads;
+        auto inst = make_instance_tbl(_ml_real);
+        check_sigmoid1_tbl(_ml, inst, id, _ppvar, _thread, _nt);
+        _r = 1.;
+        sigmoid1_tbl(_ml, inst, id, _ppvar, _thread, _nt, *getarg(1));
+        hoc_retpushx(_r);
+    }
+    static double _npy_sigmoid1(Prop* _prop) {
+        double _r{};
+        Datum* _ppvar;
+        Datum* _thread;
+        NrnThread* _nt;
+        _nrn_mechanism_cache_instance _ml_real{_prop};
+        auto* const _ml = &_ml_real;
+        size_t const id{};
+        _ppvar = _nrn_mechanism_access_dparam(_prop);
+        _thread = _extcall_thread.data();
+        _nt = nrn_threads;
+        auto inst = make_instance_tbl(_ml_real);
+        check_sigmoid1_tbl(_ml, inst, id, _ppvar, _thread, _nt);
+        _r = 1.;
+        sigmoid1_tbl(_ml, inst, id, _ppvar, _thread, _nt, *getarg(1));
+        return(_r);
+    }
+
+
+    inline int f_sigmoid1_tbl(_nrn_mechanism_cache_range* _ml, tbl_Instance& inst, size_t id, Datum* _ppvar, Datum* _thread, NrnThread* _nt, double v) {
+        int ret_f_sigmoid1 = 0;
+        inst.sig[id] = 1.0 / (1.0 + exp(inst.global->k * (v - inst.global->d)));
+        return ret_f_sigmoid1;
+    }
+
+
+    void check_sigmoid1_tbl(_nrn_mechanism_cache_range* _ml, tbl_Instance& inst, size_t id, Datum* _ppvar, Datum* _thread, NrnThread* _nt) {
```

Comment:
This function performs a lazy update, not a check. Are we free to change the name?

Reply:
Note that in NEURON there's the `nrn_thread_table_check_t` object ([link](https://github.com/neuronsimulator/nrn/blob/81ddd0edfa34dc5ce1d7ad26c63b0e3ca78691f2/src/nrnoc/membfunc.h#L43)), so I just went with that convention as coreNEURON codegen uses the same naming.

Reply:
Okay, however, please create a ticket. It can include all the other naming issues pointed out on `nmodl-references`. This is all extremely simple (linear interpolation) but not easy to read.